### ISSUE TITLE
The unit test "test_foreground_background_output" gives false positives

### ISSUE THREAD

**Issue by alalazo** [LLNL/spack#19356]
The unit test:
```
lib/spack/spack/test/llnl/util/tty/log.py::test_foreground_background_output
```
gives a fair amount of false positives on Github (failures of the test that are not due to a regression). 

### Steps to reproduce the issue
Look at failing PRs, there are high chances the failure log looks like [this one](https://github.com/spack/spack/pull/18739/checks?check_run_id=1272787694).

### Error Message
--

### Information on your system
--

### Additional information
Maintainers can retry the test suite by restarting it manually, or users can:
1. Close/reopen the PR 
2. Rebase the PR

to trigger tests again.

- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
I'm at the point where we should either fix it or xfail it, at least on macOS where it fails pretty regularly. @tgamblin wrote the original test.

**Comment by adamjstewart** [comment]
Closed by #19456
